tests/test_preprocessing.py

```diff
@@ -0,0 +1,618 @@
+        ("risk_aversion", 10, "cheese", True),
+    ],
+)
+def test_ce_from_chunk(tmp_path, recipe, eta, reduction, zero):
+    """
+    Test that ce_from_chunk returns a ce over the given chunk
+    """
+    d = tmp_path / "ce_from_chunk"
+    d.mkdir()
+    dummy_socioeconomics_dir = d / "dummy_se"
+    dummy_socioeconomics_dir.mkdir()
+    dummy_socioeconomics_file = dummy_socioeconomics_dir / "integration_dummy_se.zarr"
+
+    dummy_soecioeconomics = xr.Dataset(
+        {
+            "gdp": (
+                ["ssp", "region", "model", "year"],
+                [[[[30, 100, 30, 100, 30, 100, 30, 100, 30, 100]] * 2]] * 2,
+            ),
+            "gdppc": (
+                ["ssp", "region", "model", "year"],
+                [[[[30, 100, 30, 100, 30, 100, 30, 100, 30, 100]] * 2]] * 2,
+            ),
+            "pop": (
+                ["ssp", "region", "model", "year"],
+                [[[[30, 100, 30, 100, 30, 100, 30, 100, 30, 100]] * 2]] * 2,
+            ),
+        },
+        coords={
+            "ssp": (["ssp"], ["SSP3", "SSP4"]),
+            "region": (["region"], ["ZWE.test_region"]),
+            "model": (["model"], ["IIASA GDP", "OECD Env-Growth"]),
+            "year": (["year"], np.arange(2020, 2030)),
+        },
+    )
+
+    dummy_soecioeconomics.to_zarr(
+        dummy_socioeconomics_file, consolidated=True, mode="w"
+    )
+
+    ce_batch_coords = {
+        "ssp": np.array(["SSP3"], dtype="<U4"),
+        "region": ["ZWE.test_region"],
+        "model": np.array(["IIASA GDP"], dtype="<U15"),
+        "year": np.arange(2020, 2030),
+        "gcm": np.array(["ABCD1"], dtype="<U13"),
+        "rcp": np.array(["rcp45"], dtype="<U5"),
+    }
+
+    in_chunk = (
+        xr.Dataset(
+            {
+                "delta_dummy1": (
+                    ["gcm", "model", "rcp", "ssp", "batch", "region", "year"],
+                    [[[[[[[3, 10, 3, 10, 3, 10, 3, 10, 3, 10]]] * 15]]]],
+                ),
+                "histclim_dummy1": (
+                    ["gcm", "model", "rcp", "ssp", "batch", "region", "year"],
+                    [[[[[[[30, 100, 30, 100, 30, 100, 30, 100, 30, 100]]] * 15]]]],
+                ),
+            },
+            coords={
+                "gcm": (["gcm"], ["ABCD1"]),
+                "model": (["model"], ["IIASA GDP"]),
+                "rcp": (["rcp"], ["rcp45"]),
+                "ssp": (["ssp"], ["SSP3"]),
+                "batch": (["batch"], np.arange(15)),
+                "region": (["region"], ["ZWE.test_region"]),
+                "year": (["year"], np.arange(2020, 2030)),
+            },
+        )
+        .astype(np.float32)
+        .chunk(
+            {
+                "batch": 15,
+                "ssp": 1,
+                "model": 1,
+                "rcp": 1,
+                "gcm": 1,
+                "year": 10,
+                "region": 24378,
+            }
+        )
+    )
+
+    if reduction not in ["cc", "no_cc"]:
```

Comment:
@JMGilbert this is another spot where the test is testing two things. This conditional testing of the error should be another test function.

Reply:
For sure -- I was somewhat reluctant to separate a few of these tests because the conditional here still takes some of the synthetic inputs that I generate above this line. I was considering doing a lot of generalization (because I learned about how to make tests share inputs later on), but wasn't sure how much extra time to put in to clean these tests up given that the functionality won't actually change. I could also just copy and paste the synthetic inputs and this conditional into a new test but that feels wrong.

Reply:
(happy to do either, just wanted to make sure that you want the time put in on this)

Reply:
I see what you mean. I think going the easy route and copying the synthetic inputs to another test func is probably fine - what do you think @brews ?

Reply:
@kemccusker I think spare copy-and-paste is fine for an early test for legacy code — like, don't make it too much of a habit. Doing a nice pytest fixture to help setup input data or something might be a better way of doing it, but it's not worth the effort if we do it sloppy or if we're likely to turn around and refactor the code we are testing anyways. I'd let y'all be the judge of that and if its worth cleaning up the test so you can read its intentions later. I would argue the important thing for now, and this PR, is to get *some* kind of test in place for this code/behavior.

Reply:
ok @JMGilbert let's go w/ a new test the "easy" way for now since it should be quick, keeping in mind Brewster's advice for future refactors.